File: src/easy/IsBoomerang.swift
func isBoomerang(_ points: [[Int]]) -> Bool {
  var distinct = Set<String>()
  for point in points {
    distinct.insert("\(point[0]),\(point[1])")
  }
  if distinct.count != 3 {
    return false
  }
let x1 = points[0][0], y1 = points[0][1]
    let x2 = points[1][0], y2 = points[1][1]
      let x3 = points[2][0], y3 = points[2][1]
let a = x1*(y2-y3)
let b = x2*(y3-y1)
let c = x3*(y1-y2)
  	return a + b + c != 0
}

File: src/main/java/easy/IsBoomerang.java
package easy;

import java.util.HashSet;

public class IsBoomerang {
  public boolean solve(int[][] points) {
    HashSet<String> distinct = new HashSet<>();
    for (int[] point : points) {
      String key = String.format("%d,%d", point[0], point[1]);
      if (distinct.contains(key)) return false;
      distinct.add(key);
    }
    int x1 = points[0][0];
    int y1 = points[0][1];
    int x2 = points[1][0];
    int y2 = points[1][1];
    int x3 = points[2][0];
    int y3 = points[2][1];
    return Math.abs(x1 * (y2 - y3) + x2 * (y3 - y1) + x3 * (y1 - y2)) != 0;
  }
}
